feat(oxfmt): Arrange cli mode and update help

leaysgur opened the pull request:
Updated to:

```
Usage: oxfmt [-c=PATH] [PATH]...

Mode Options:
        --init               Initialize `.oxfmtrc.jsonc` with default values
        --lsp                Start language server protocol (LSP) server

Output Options:
        --write              Format and write files in place (default)
        --check              Check if files are formatted, also show statistics
        --list-different     List files that would be changed

Config Options
    -c, --config=PATH        Path to the configuration file

Ignore Options
        --ignore-path=PATH   Path to ignore file(s). Can be specified multiple times. If not
                             specified, .gitignore and .prettierignore in the current directory are
                             used.
        --with-node-modules  Format code in node_modules directory (skipped by default)

Runtime Options
        --no-error-on-unmatched-pattern  Do not exit with error when pattern is unmatched
        --threads=INT        Number of threads to use. Set to 1 for using only 1 CPU core.

Available positional items:
    PATH                     Single file, single path or list of paths. If not provided, current
                             working directory is used. Glob is supported only for exclude patterns
                             like `'!**/fixtures/*.js'`.

Available options:
    -h, --help               Prints help information
    -V, --version            Prints version information

```

leaysgur commented:
* **#16728** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/16728?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/16728?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=16728).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen reviewed (changes requested):
"Available positional items" should be at the top, otherwise it's hard to find what PATH is.

leaysgur committed:
feat(oxfmt): Arrange cli mode and update help (#16728)

Updated to:

```
Usage: oxfmt [-c=PATH] [PATH]...

Mode Options:
        --init               Initialize `.oxfmtrc.jsonc` with default values
        --lsp                Start language server protocol (LSP) server

Output Options:
        --write              Format and write files in place (default)
        --check              Check if files are formatted, also show statistics
        --list-different     List files that would be changed

Config Options
    -c, --config=PATH        Path to the configuration file

Ignore Options
        --ignore-path=PATH   Path to ignore file(s). Can be specified multiple times. If not
                             specified, .gitignore and .prettierignore in the current directory are
                             used.
        --with-node-modules  Format code in node_modules directory (skipped by default)

Runtime Options
        --no-error-on-unmatched-pattern  Do not exit with error when pattern is unmatched
        --threads=INT        Number of threads to use. Set to 1 for using only 1 CPU core.

Available positional items:
    PATH                     Single file, single path or list of paths. If not provided, current
                             working directory is used. Glob is supported only for exclude patterns
                             like `'!**/fixtures/*.js'`.

Available options:
    -h, --help               Prints help information
    -V, --version            Prints version information

```

pacak commented:
>> "Available positional items" should be at the top, otherwise it's hard to find what PATH is.

> @Boshen Unfortunately, this seems bpaf's design. (And current output also the same)
> https://github.com/pacak/bpaf/blob/5aef3d7defbcbe5480590ee948133161fe79a02f/src/lib.rs#L1374

Link refers to a different problem. In current version positional items must be consumed last (placed at the end of the parser) and changing this is hard without changing the design. Items in the help menu are sorted according to some rules and ability to move positional items higher is not exposed, but not really blocked by the design.


Speaking about changing the design. I'm working on a next version that removes some of the restrictions. Still some time away. Are there any features you wanted to have but couldn't achieve with current API?
